lib/galaxy/security/validate_user_input.py

```diff
@@ -164,3 +173,35 @@ def validate_preferred_object_store_id(
     trans, object_store: ObjectStore, preferred_object_store_id: Optional[str]
 ) -> str:
     return object_store.validate_selected_object_store_id(trans.user, preferred_object_store_id) or ""
+
+
+def is_email_banned(email: str, filepath: Optional[str]) -> bool:
+    if not filepath:
+        return False
+    email = _make_canonical_email(email)
+    banned_emails = _read_email_ban_list(filepath)
+    for address in banned_emails:
+        if email == _make_canonical_email(address):
+            return True
+    return False
+
+
+def _make_canonical_email(email: str) -> str:
+    """
+    Transform to canonical representation:
+    - lowercase
+    - gmail: drop periods in local-part
+    - gmail: drop plus suffixes in local-part
+    """
+    email = email.lower()
+    localpart, domain = email.split("@")
+    if domain == "gmail.com":
+        localpart = localpart.replace(".", "")
+        if localpart.find("+") > -1:
```

Comment:
plus addressing / sub addressing is not a gmail exclusive feature. This works on many mail servers